### ISSUE TITLE
custom_status_bar: feature request: add option for remaining to show "negatives" / late

### ISSUE THREAD

**Issue by leftos** [cgtrout/MSFS-PyScriptManager#25]
Right now the remaining time clamps to 0, which generally makes sense, but it would be nice to have an option to have it not be clamped so I can see how late I am currently vs my expected arrival time.

Another option is to just show the STA (scheduled time of arrival) when remaining reaches 0.

A few examples of the above ideas:

Option 1
With time remaining: `Remaining: 00:15:23`
With no time remaining: `Remaining: -00:03:22`

Option 2 (a bit more formatted to be user-friendly):
With time remaining: `Remaining: 00:15:23`
With no time remaining: `Remaining: 3m late`

Option 3 (just show ETA):
With time remaining: `Remaining: 00:15:23`
With no time remaining: `Remaining: Late (STA 1523z)`

**Comment by cgtrout** [comment]
Hi leftos:

Thanks for the idea!  I will look into adding this later.  My thought is to add a new form that will show all time related settings (opens on double click).  This will give a streamlined way to change settings (including simbrief settings) for the countdown timer all on one page.

I also added an issue myself with some of my other planned updates for custom_status_bar. (https://github.com/cgtrout/MSFS-PyScriptManager/issues/17)

At the moment my priority is going to be getting the next release ready.  Once that is done then I'll start work on the custom_status_bar enhancements.

Thanks for the continued feedback!

**Comment by leftos** [comment]
Totally, no rush on this at all. Thanks for being open to my suggestions and feedback!

**Comment by cgtrout** [comment]
Hi leftos:

This feature has been implemented as part of the changes I've been making.  I was originally going to delay the custom_status_bar changes until I pushed out a new release but decided to work on this first.

Now when the bar is double-clicked it will bring up a custom dialog:
![image](https://github.com/user-attachments/assets/72ab39ad-908a-49f0-8992-44ba88aee310)

This allows the timer to be set as before, but also presents a collapsible page to configure the SimBrief connectivity.

The drop down next to "Select SimBrief Time" allows the user to select either the "Estimated In" or "Estimated TOD" time from the flight plan.

Gate Out Time - this allows user to set a different gate out time if ahead/behind schedule.

Settings on this page are saved in the /Settings/custom_status_bar.json file.

For now the user must click "Get Time" to load a time into the timer but if requested I can likely fairly easily add full automation back to the script.

All of the changes have been merged with new-version-update(https://github.com/cgtrout/MSFS-PyScriptManager/tree/new-version-update).  You can update to this to try it out or if you prefer you can wait until the next release is out.  The only remaining thing I want to add is multi-template support that loads from the setting file - that should hopefully be done soon.

Due to the high amounts of changes, there may be some bugs in the script, but my testing puts it in a good place.  Only bug I'm aware of is that it will show a large negative value at load if "Allow Negative Timer" is enabled.

**Comment by leftos** [comment]
I think it would be nice to have full automation optionally as something that can be checked in that new Simbrief settings dialog, if we're considering it!

Either way, awesome stuff here, can't wait to try it out!

**Comment by cgtrout** [comment]
Ability to automate Simbrief timer functionality has been added.(https://github.com/cgtrout/MSFS-PyScriptManager/pull/30)

**Comment by cgtrout** [comment]
Mult-template support has been added as well(https://github.com/cgtrout/MSFS-PyScriptManager/commit/04aba1018fdb3fe05c48860ac9da1963285540bd)

For this, it will now load a template from /Settings/status_bar_templates.py.  It is a bit unconventional to load from a python file for this, but I thought this will work well (relatively easy to modify).

Note that this hasn't been tested on my sim-computer.  It will be several days until I can test this last change with the SIM although I don't anticipate this breaking anything from before.

**Comment by cgtrout** [comment]
One more thing to add:
I've added some documentation for this - not sure how intuitive it is to use in its current form so I thought I'd mention it quickly:
https://github.com/cgtrout/MSFS-PyScriptManager/blob/new-version-update/Docs/custom_status_bar.md

I've also added credit for your ideas/contributions in this file.

I'll leave it at this for now, but if you have any feedback on any of the above changes feel free to let me know - thanks!

**Comment by leftos** [comment]
Awesome stuff, no notes from me for now.

**Comment by cgtrout** [comment]
Thanks, will consider this issue closed for now, but again feel free to reply or create new issues.
